quarkchain/cluster/master.py

```diff
@@ -682,6 +683,20 @@ def __get_mining_params():
             guardian_private_key=self.env.quark_chain_config.guardian_private_key,
         )
 
+    async def __rebroadcast_committing_root_block(self):
+        committing_block_hash = self.root_state.get_committing_block_hash()
+        if committing_block_hash:
+            r_block = self.root_state.db.get_root_block_by_hash(
+                committing_block_hash, consistency_check=False
+            )
+            future_list = self.broadcast_rpc(
```

Comment:
If we move committing_block_hash() before add_block() (see below), r_block may be None and we should just skip re-broadcasting.  If r_block is not None, a log message may be preferred.